### PR TITLE
Basic super classes

### PR DESCRIPTION
**Pull request by tkeeler**
I have a basic outline of the super classes for the timeline related objects.
I added 3 classes that break down as follows.
Likeable extends model and is only for classes that can be liked and not commented on (i.e. comments)
Commentable extends Likeable and is for classes that can be liked and comment on (should be everything else)
Status extends Commentable and is a container class for all the other objects for time line aggregation. This class may be redundant, but does offer a simple text field for each of the objects and a pointer to another. Date date is gone and has been replaced with createdAt and updatedAt from Model.java

Known Issues:
No UI to add to these yet.

Features to add:
A true time line controller that will grab and display a users time line.
UI for adding comments, likes, and statuses.

### CODE REVIEW

File: app/controllers/Comments.java [hunk 1]
```diff
@@ -12,14 +12,9 @@
 	
 	//comments(Long id): will render the user being viewed unless it is a null user then it will render the current user
 	public static void comments(Long id) {
-	    User user = id == null ? user() : (User) User.findById(id);
+	    User user = id == null ? Application.user() : (User) User.findById(id);
```
Comment (amshali) [inline app/controllers/Comments.java:15]: OBController already has user() in it. Use that.

Comment (tkeeler) [inline app/controllers/Comments.java:15]: Made the change.

Comment (amshali) [inline app/controllers/Comments.java:15]: did you push?

Comment (tkeeler) [inline app/controllers/Comments.java:15]: I pushed to both my master and Status branches. Where else do I need to push this to?

**Pull request merged**